### PR TITLE
Update doc about installation on different versions of ubuntu and debian

### PR DESCRIPTION
**Pull request by saraansh1999**
Added clarifications for building from source for older versions of Ubuntu and Debian.

### CODE REVIEW

File: doc/guide/build.hpp [hunk 1]
```diff
@@ -104,6 +105,17 @@ In Ubuntu and Debian, you can get all of these dependencies through apt:
...
 
+@note For older versions of Ubuntu, Armadillo needs to be built from 
+source as apt installs an older version. So you need to omit 
+\c libarmadillo-dev from the code snippet above. Then 
+<a href="http://arma.sourceforge.net/download.html">this link</a>
+can be used to download the file and the README in the uncompressed folder
+can be followed to perform the build. <br> <br>
+Also note that if apt is not able to find \c libensmallen-dev and \c 
+libstb-dev, they are downloaded and installed automatically by 
+CMake if you keep the flags \c DOWNLOAD_ENSMALLEN and \c DOWNLOAD_STB_IMAGE
```
Comment (birm) [inline doc/guide/build.hpp]: I think that the stb and enemallen cmake options are well described above (lines 83 and 87-88).

File: doc/guide/build.hpp [hunk 2]
```diff
@@ -95,7 +95,8 @@ For Python bindings, the following packages are required:
  - pandas >= 0.15.0
  - pytest-runner
 
-In Ubuntu and Debian, you can get all of these dependencies through apt:
+In Ubuntu(>= 18.04) and Debian all of these dependencies can be
```
Comment (birm) [inline doc/guide/build.hpp]: Is there also a minimum debian version?

File: doc/guide/build.hpp [hunk 3]
```diff
@@ -104,6 +105,17 @@ In Ubuntu and Debian, you can get all of these dependencies through apt:
   libstb-dev
 @endcode
 
+@note For older versions of Ubuntu, Armadillo needs to be built from 
+source as apt installs an older version. So you need to omit 
+\c libarmadillo-dev from the code snippet above. Then 
+<a href="http://arma.sourceforge.net/download.html">this link</a>
+can be used to download the file and the README in the uncompressed folder
```
Comment (birm) [inline doc/guide/build.hpp]: I'd reword this just a little. Maybe something like "Download and extract the <a href="http://arma.sourceforge.net/download.html">armadillo source</a>, and follow the instructions to build and install it."

Comment (saraansh1999) [reply doc/guide/build.hpp]: Sure I'll do that.

File: doc/guide/build.hpp (hunk 2 above)
Comment (saraansh1999) [reply doc/guide/build.hpp]: I'm not sure about that. Would love input from some Debian user or i can checkout the documentations if no one is available by tomorrow.

File: doc/guide/build.hpp (hunk 1 above)
Comment (saraansh1999) [reply doc/guide/build.hpp]: Yeah my bad. I missed those few points during my read. Should i remove this?

Comment (saraansh1999) [reply doc/guide/build.hpp]: I added these as the snippet given was generating an error and many people like me would directly copy paste the snippet and might be confused. So i wrote the line with the exact problem that might occur.

Comment (birm) [reply doc/guide/build.hpp]: I think it's best to take them out; we also have #2235 open, which removes ensmallen and stb from this apt install in this same document.

**Comment by saraansh1999** [comment]
Are all the checks performed every time a commit is made?

File: doc/guide/build.hpp (hunk 2 above)
Comment (birm) [reply doc/guide/build.hpp]: See https://packages.debian.org/source/buster/armadillo ; it looks like "buster" (10) and "bullseye" (11) (and "sid" (unstable)) have sufficient armadillo.

**Comment by birm** [comment]
@saraansh1999 yes, the checks are run after each commit.

**Comment by saraansh1999** [comment]
How long does this usually take? It's been stuck on 3 pending tests for a long time.

File: doc/guide/build.hpp [hunk 4]
```diff
@@ -95,14 +95,22 @@ For Python bindings, the following packages are required:
  - pandas >= 0.15.0
  - pytest-runner
 
-In Ubuntu and Debian, you can get all of these dependencies through apt:
+In Ubuntu (>= 18.04) and Debian (buster, bullseye, sid) all of these dependencies can be
```
Comment (birm) [inline doc/guide/build.hpp]: please use debian version numbers (>=10.0)

**Comment by birm** [comment]
Yes, many of these builds are done one-at-a-time due to the resources we have. Luckily, your PR only touches documentation, and no tested code, so we don't have to wait for results

**Comment by saraansh1999** [comment]
So I have already made all the requested changes. Is there anything else that i need to do?

**Comment by saraansh1999** [comment]
I have made the requested changes. Please tell me if there is anything more to do.
Thanks for the inputs.

File: doc/guide/build.hpp [hunk 5]
```diff
@@ -95,14 +95,22 @@ For Python bindings, the following packages are required:
...
   python-pandas python-numpy cython python-setuptools libensmallen-dev
   libstb-dev
 @endcode
+ 
+@note For older versions of Ubuntu and Debian , Armadillo needs to be built from 
+source as apt installs an older version. So you need to omit 
+\c libarmadillo-dev from the code snippet above and instead use
+<a href="http://arma.sourceforge.net/download.html">this link</a>
+ to download the required file. Extract this file and follow the README in the 
+ uncompressed folder to build and install armadillo.
```
Comment (rcurtin) [inline doc/guide/build.hpp]:
```suggestion
 uncompressed folder to build and install Armadillo.
```

File: doc/guide/build.hpp [hunk 6]
```diff
@@ -95,14 +95,22 @@ For Python bindings, the following packages are required:
  - pandas >= 0.15.0
  - pytest-runner
 
-In Ubuntu and Debian, you can get all of these dependencies through apt:
+In Ubuntu (>= 18.04) and Debian (>= 10) all of these dependencies can be 
+installed through apt :
```
Comment (rcurtin) [inline doc/guide/build.hpp]:
```suggestion
installed through apt:
```

**Comment by saraansh1999** [comment]
@rcurtin I have updated the PR with the requested changes.

File: doc/guide/build.hpp [hunk 7]
```diff
@@ -95,14 +95,22 @@ For Python bindings, the following packages are required:
...
+installed through apt:
 
 @code
 # apt-get install libboost-math-dev libboost-program-options-dev
   libboost-test-dev libboost-serialization-dev libarmadillo-dev binutils-dev
   python-pandas python-numpy cython python-setuptools libensmallen-dev
   libstb-dev
 @endcode
+ 
+@note For older versions of Ubuntu and Debian , Armadillo needs to be built from
```
Comment (zoq) [inline doc/guide/build.hpp]: Actually it looks like we can remove the extra space before `.`.

**Comment by saraansh1999** [comment]
@zoq I'm not sure as to where exactly you are pointing to. Could you please clarify the request?

File: doc/guide/build.hpp (hunk 7 above)
Comment (zoq) [reply doc/guide/build.hpp]: `Debian , Armadillo` -> `Debian, Armadillo`

**Comment by saraansh1999** [comment]
Are there any other required changes?

**Comment by saraansh1999** [comment]
I already got the stickers message

**Comment by rcurtin** [comment]
Ok, awesome, got it working.  Thanks for the contribution! :+1:

**Comment by saraansh1999** [comment]
When will this change reflect online?

**Comment by rcurtin** [comment]
The `mlpack-git` documentation gets rebuilt nightly, so in about 12 hours or so the changes should be reflected on https://www.mlpack.org/doc/mlpack-git/doxygen/build.html. :+1:
